tests/components/bsblan/test_water_heater.py

```diff
@@ -0,0 +1,253 @@
+"""Tests for the BSB-Lan water heater platform."""
+
+from datetime import timedelta
+from unittest.mock import AsyncMock, MagicMock
+
+from bsblan import BSBLANError, StaticState
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.bsblan.const import DOMAIN
+from homeassistant.components.water_heater import (
+    DOMAIN as WATER_HEATER_DOMAIN,
+    STATE_ECO,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_TEMPERATURE, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.entity_registry as er
+
+from . import setup_with_selected_platforms
+
+from tests.common import (
+    MockConfigEntry,
+    async_fire_time_changed,
+    load_json_object_fixture,
+    snapshot_platform,
+)
+
+ENTITY_ID = "water_heater.bsb_lan"
+
+
+@pytest.fixture
+def mock_bsblan_with_methods(mock_bsblan):
+    """Create a mock with proper method signatures."""
+    # Create an async mock for set_hot_water with the correct signature
+    mock_bsblan.set_hot_water = AsyncMock()
+    return mock_bsblan
```

Comment:
Not sure why t his is here, you use autospec in the creation of the mock, so the field is already existing and is an AsyncMock

Reply:
lol this is indeed not good. When I fixed it I found a bug the bsblan missed the set operation_mode. I removed it (not sure for what reason) and did not put it back. So I need first to update the python lib again :( thanks  for finding this issue